articles/152_roslaunch_multi_machine_launch.md

```diff
@@ -0,0 +1,310 @@
+5d186778-1f50-4828-9425-64cc2ed1342c
+$
+```
+
+```bash
+$ ros2 launcher list
+ab1e0138-bb22-4ec9-a590-cf377de42d0f: 5 nodes, 2 hosts
+50bda6fb-d451-4d53-8a2b-e8fcdce8170b: 2 nodes, 1 host
+5d186778-1f50-4828-9425-64cc2ed1342c: 16 nodes, 3 hosts
+$
+```
+
+#### `attach`
+
+Since it is possible to detach from a launched system, it is useful for
+scripting or diagnostic purposes to be able to re-attach to it.
+
+```bash
+$ ros2 launcher attach -h
+usage: ros2 launcher attach [-h] [-v] [--spin-time SPIN_TIME] [system_id]
+
+Blocks until all nodes running under the specified Launch System ID have exited
+
+positional arguments:
+  system_id          Launch System ID of the nodes to attach to; if less than
+                     a full UUID is specified, it will attach to the first
+                     Launch System it finds whose ID begins with that sub-string
+
+optional arguments:
+  -h, --help            Show this help message and exit.
+  -v, --verbose         Provides more verbose output.
+  --spin-time SPIN_TIME
+                        Spin time in seconds to wait for discovery (only
+                        applies when not using an already running daemon)
+```
+
+Example output:
+
+```bash
+$ ros2 launcher attach 50bda6fb-d451-4d53-8a2b-e8fcdce8170b
+Attached to Launch System 50bda6fb-d451-4d53-8a2b-e8fcdce8170b.
+(... in another terminal, run `ros2 launcher term 50bda6fb`...)
+All nodes in Launch System 50bda6fb-d451-4d53-8a2b-e8fcdce8170b have exited.
+$
+```
+
+Verbose mode:
+
+```bash
+$ ros2 launcher attach -v 50bda6fb
+Attached to Launch System 50bda6fb-d451-4d53-8a2b-e8fcdce8170b.
+Waiting for node /launch_ros
+Waiting for node /talker
+Waiting for node /listener
+(... in another terminal, run `ros2 launcher term 50bda6fb`...)
+Node /launch_ros has exited
+Node /talker has exited
+Node /listener has exited
+All nodes in Launch System 50bda6fb-d451-4d53-8a2b-e8fcdce8170b have exited.
+$
+```
+
+#### `term`
+
+Terminates all nodes that were launched under a specific Launch System ID.
+
+```bash
+$ ros2 launcher term -h
+usage: ros2 launcher term [-h] [-v] [--spin-time SPIN_TIME] [system_id]
+
+Terminates all nodes that were launched under a specific Launch System ID
+
+positional arguments:
+  system_id          Launch System ID of the nodes to terminate; if less than
+                     a full UUID is specified, it will terminate nodes
+                     belonging to the first Launch System it finds whose ID
+                     begins with that sub-string
+
+optional arguments:
+  -h, --help            Show this help message and exit.
+  -v, --verbose         Provides more verbose output.
+  --spin-time SPIN_TIME
+                        Spin time in seconds to wait for discovery (only
+                        applies when not using an already running daemon)
+```
+
+Example output:
+
+```bash
+$ ros2 launcher term 50bda6fb-d451-4d53-8a2b-e8fcdce8170b
+Terminating Launch System 50bda6fb-d451-4d53-8a2b-e8fcdce8170b.
+$
+```
```

Comment:
nit: newline

Reply:
Note to resolve this comment: EOF newlines are present unless stated otherwise in github viewer